tools/cli/lib/api.c

```diff
@@ -678,192 +684,166 @@ TDNFCliRepoSyncCommand(
     goto cleanup;
 }
 
-uint32_t tag_handler(const char *tag, void *data, char **pszRet)
+static uint32_t tag_handler(const char *tag, void *data, char **pszRet)
 {
     PTDNF_PKG_INFO pPkgInfo = (PTDNF_PKG_INFO)data;
     char *pszVal = NULL;
-    int depkey=0;
-    unsigned int dep_flag=0;
     uint32_t dwError = 0;
 
-    if (strcmp(tag, "name") == 0) {
-        pszVal = pPkgInfo->pszName;
-    } else if (strcmp(tag, "arch") == 0) {
-        pszVal = pPkgInfo->pszArch;
-    } else if (strcmp(tag, "version") == 0) {
-        pszVal = pPkgInfo->pszVersion;
-    } else if (strcmp(tag, "reponame") == 0) {
-        pszVal = pPkgInfo->pszRepoName;
-    } else if (strcmp(tag, "release") == 0) {
-        pszVal = pPkgInfo->pszRelease;
-    } else if (strcmp(tag, "evr") == 0) {
-        pszVal = pPkgInfo->pszEVR;
-    } else if (strcmp(tag, "sourcename") == 0) {
-        pszVal = pPkgInfo->pszSourcePkg;
-    } else if (strcmp(tag, "size") == 0) {
-        pszVal = pPkgInfo->pszFormattedDownloadSize;
-    } else if (strcmp(tag, "downloadsize") == 0) {
-        pszVal = pPkgInfo->pszFormattedDownloadSize;
-    } else if (strcmp(tag, "installsize") == 0) {
-        pszVal = pPkgInfo->pszFormattedSize;
-    } else if (strcmp(tag, "sourcerpm") == 0) {
-        pszVal = pPkgInfo->pszSourcePkg;
-    } else if (strcmp(tag, "description") == 0) {
-        pszVal = pPkgInfo->pszDescription;
-    } else if (strcmp(tag, "summary") == 0) {
-        pszVal = pPkgInfo->pszSummary;
-    } else if (strcmp(tag, "license") == 0) {
-        pszVal = pPkgInfo->pszLicense;
-    } else if (strcmp(tag, "url") == 0) {
-        pszVal = pPkgInfo->pszURL;
-    } else {
-        for (depkey = 0; depkey < REPOQUERY_DEP_KEY_COUNT; depkey++)
-        {
-            if (strcmp(tag, depKey[depkey]) == 0)
-            {
+    typedef struct _TagMapping_t {
+        const char *tag;
+        char *member;
+    } TagMapping_t;
+
+    TagMapping_t tagMappings[] = {
+        {"name", pPkgInfo->pszName},
+        {"arch", pPkgInfo->pszArch},
+        {"version", pPkgInfo->pszVersion},
+        {"reponame", pPkgInfo->pszRepoName},
+        {"release", pPkgInfo->pszRelease},
+        {"evr", pPkgInfo->pszEVR},
+        {"sourcename", pPkgInfo->pszSourcePkg},
+        {"size", pPkgInfo->pszFormattedDownloadSize},
+        {"downloadsize", pPkgInfo->pszFormattedDownloadSize},
+        {"installsize", pPkgInfo->pszFormattedSize},
+        {"sourcerpm", pPkgInfo->pszSourcePkg},
+        {"description", pPkgInfo->pszDescription},
+        {"summary", pPkgInfo->pszSummary},
+        {"license", pPkgInfo->pszLicense},
+        {"url", pPkgInfo->pszURL},
+        {NULL, NULL}
+    };
+
+    for (int i = 0; tagMappings[i].tag; ++i) {
+        if (!strcmp(tag, tagMappings[i].tag)) {
+            pszVal = tagMappings[i].member;
+            *pszRet = strdup(pszVal);
+            return dwError;
```

Comment:
Remove this line and the `strdup()` above - `pszVal` will be copied below (line 739) if set.